### PR TITLE
`Development`: Add CORS allowed-origin-patterns

### PR DESCRIPTION
**Pull request by janthoXO**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Server
- [x] **Important**: I implemented the changes with a [very good performance](https://docs.artemis.cit.tum.de/dev/guidelines/performance/) and prevented too many (unnecessary) and too complex database calls.
- [x] I **strictly** followed the principle of **data economy** for all database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I documented the Java code using JavaDoc style.

### Description
Fixes CORS so that configured `allowed-origin-patterns` are also accepted.
Furthermore adds the VSCode plugin origins to the CORS in `application.yml`

### Steps for Testing

only Config changes so code reviews are appreciated

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

## Summary by CodeRabbit

- **New Features**
	- Enhanced CORS filter configuration to support a broader range of allowed origins and patterns for API endpoints.

- **Bug Fixes**
	- Improved logic for registering the CORS filter to ensure proper handling of origin patterns.

@coderabbitai pause

### CODE REVIEW

**Review by iyannsch — APPROVED** [review]
Code LGTM, we should maybe test if the App etc are still able to fetch all their data

**Comment by janthoXO** [comment]
> Code LGTM, we should maybe test if the App etc are still able to fetch all their data

yes that would be good. currently cors is disabled on production and testservers. so we need the cors adjustment on the test servers first

**Review by SimonEntholzer — DISMISSED** [review]
Code lgtm. How could I test this locally?

**Comment by janthoXO** [comment]
> Code lgtm. How could I test this locally?

its hard to test. basically you would need a second website with a different URL that makes requests to the server. and then you would set the allowed-origin-patterns in the application.yml

File: src/main/resources/config/application.yml [hunk 1]
```diff
@@ -284,13 +284,16 @@ jhipster:
     clientApp:
         name: 'artemisApp'
     # By default CORS is disabled. Uncomment to enable.
```
Comment (magaupp) [inline src/main/resources/config/application.yml:286]: Nitpick: Remove the outdated comment

File: src/main/java/de/tum/cit/aet/artemis/core/config/WebConfigurer.java [hunk 2]
```diff
@@ -117,7 +117,8 @@ private String resolvePathPrefix() {
     public CorsFilter corsFilter() {
         UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
         CorsConfiguration config = jHipsterProperties.getCors();
-        if (config.getAllowedOrigins() != null && !config.getAllowedOrigins().isEmpty()) {
+        if ((config.getAllowedOrigins() != null && !config.getAllowedOrigins().isEmpty())
+            || (config.getAllowedOriginPatterns() != null && !config.getAllowedOriginPatterns().isEmpty())) {
```
Comment (magaupp) [inline src/main/java/de/tum/cit/aet/artemis/core/config/WebConfigurer.java:121]: Consider using CollectionUtils.isEmpty() like JHipster's newer template does.
```suggestion
        if (!CollectionUtils.isEmpty(config.getAllowedOrigins()) || !CollectionUtils.isEmpty(config.getAllowedOriginPatterns())) {
```

**Review by magaupp — APPROVED** [review]
Code LGTM

File: src/main/resources/config/application.yml (hunk 1 above)
Comment (janthoXO) [reply src/main/resources/config/application.yml]: changed the comment thanks

File: src/main/java/de/tum/cit/aet/artemis/core/config/WebConfigurer.java
Comment (magaupp) [inline src/main/java/de/tum/cit/aet/artemis/core/config/WebConfigurer.java:1]: An import is missing:
import org.springframework.util.CollectionUtils;

**Review by magaupp — CHANGES_REQUESTED** [review]
Does not build.

**Review by iyannsch — DISMISSED** [review]
Code LGTM as long as the origins are authentic

**Review by magaupp — DISMISSED** [review]
Code LGTM

**Review by SimonEntholzer — APPROVED** [review]
lgtm

**Review by iyannsch — APPROVED** [review]

**Review by magaupp — APPROVED** [review]

**Review by eceeeren — APPROVED** [review]
Code LGTM

**Pull request merged**